### ISSUE TITLE
[Bug] Linter does not highlight errors as shown in the docs

### ISSUE THREAD

**Issue by julior-ox** [julia-vscode/julia-vscode#2881]
##  Behavior, as shown in docs
https://github.com/julia-vscode/julia-vscode/wiki/Information#linter:

![image](https://user-images.githubusercontent.com/104969766/166878141-028ccb15-9f00-471f-b75f-04d7244f0279.png)

https://www.julia-vscode.org/:

![image](https://user-images.githubusercontent.com/104969766/166878268-2a2319b6-7730-4440-a658-b331d6bc93f4.png)

## Actual behavior
![actual](https://user-images.githubusercontent.com/104969766/166878763-1ed54b02-d030-4e11-9834-ea78048b6eea.png)

Expected: 
red error at `x` -> undeclared symbol
yellow error at `maxi`, hinting to `maxiter`

## Description

Apparently with current version, the integrated linter does not highlight undeclared symbol errors anymore. Are  there any additional steps to do? Has there been a  change in behavior for recent Julia VS Code versions?

## Steps to reproduce

- Download fresh new VS Code and Julia versions:
```
curl -L -O -J "https://code.visualstudio.com/sha/download?build=stable&os=linux-rpm-x64"
sudo dnf install code-1.66.2-1649664637.el7.x86_64.rpm
curl -L -O -J "https://julialang-s3.julialang.org/bin/linux/x64/1.7/julia-1.7.2-linux-x86_64.tar.gz"
# Install Julia 1.6.17 extension
```

- Generate sample `test_proj` project:
```
(@v1.7) pkg> generate test_proj
  Generating  project test_proj:
    test_proj/Project.toml
    test_proj/src/test_proj.jl

[user@test-julia ~]$ code test_proj/
```

- ensure linter is enabled and julia exec path set in `settings.json`:
```json
{
    "julia.executablePath": "/home/user/julia-1.7.2/bin/julia",
    "terminal.integrated.commandsToSkipShell": [
        "language-julia.interrupt"
    ],
    "julia.symbolCacheDownload": true,
    "julia.enableTelemetry": false,
    "julia.lint.run": true
}
```

- Activate julia project environment

- Output from julia language server seems normal:
> [ Info: Starting the Julia Language Server
> [ Info: Symbol server store is at '/home/user/.config/Code/User/globalStorage/julialang.language-julia/symbolstorev5'.
> [ Info: Starting LS at 1651733441
> [ Info: Received new data from Julia Symbol Server.
> [ Info: Received new data from Julia Symbol Server.

**Comment by julior-ox** [comment]
After further investigation I found the extension setting `julia.lint.missingrefs`. Setting it to `all` via
```
"julia.lint.missingrefs": "all"
```
now properly display an error message for above code:
> Missing reference: x

I guess this is the same behavior as shown in the docs with message "use of undeclared symbol", just reworded?

My suggestion: set [default value](https://github.com/julia-vscode/julia-vscode/blob/818a02740c897e140a195aaa74d633f1f1ffc201/package.json#L848) to `all` instead of `none` to make experience more newcomer-friendly.

Would anything speak against this change?

**Comment by pfitzseb** [comment]
Yes, that linter pass is very prone to false positives, so we decided to use this default until we're more confident in the results.

**Comment by julior-ox** [comment]
Thanks for the info!
I did not have much codebase with Julia yet. The only thing I noticed was a false positive with `@enum` as discussed [here](https://discourse.julialang.org/t/missing-reference-warnings-with-julia-in-vs-code/31321).

I still think error highlighting of missing references far outweighs other edge cases - e.g. assuming majority of newcomers does not use `@enum`. Everyone needs and knows this feature from languages/VS Code extensions as Python.

Though you surely know better other more annoying false positives.

**Comment by pfitzseb** [comment]
Yeah, that's fair. The proper fix for this is of course to fix the false positives...

**Comment by xgdgsc** [comment]
When will we be confident? I feel some false positive is better than silly bugs that happen at runtime, which could have been spotted with this easily.

**Comment by xgdgsc** [comment]
It seems when `all` is set. The completion slows down too much as `func` completion to `function` even hangs for a long time. Is `symbols` a good compromise?
